dot/state/service_test.go

```diff
@@ -128,3 +129,72 @@ func TestService_BlockTree(t *testing.T) {
 		t.Fatalf("Fail: got %s expected %s", stateA.Block.BestBlockHash(), stateB.Block.BestBlockHash())
 	}
 }
+func Test_ServicePruneStorage(t *testing.T) {
+	testDir := utils.NewTestDir(t)
+	defer utils.RemoveTestDir(t)
+
+	serv := NewService(testDir, log.LvlTrace)
+	serv.UseMemDB()
+
+	genesisHeader, err := types.NewHeader(common.NewHash([]byte{0}), big.NewInt(0), trie.EmptyHash, trie.EmptyHash, [][]byte{})
+	require.Nil(t, err)
+
+	genesisData := new(genesis.Data)
+
+	tr := trie.NewEmptyTrie()
+	err = serv.Initialize(genesisData, genesisHeader, tr, firstEpochInfo)
+	require.Nil(t, err)
+
+	err = serv.Start()
+	require.Nil(t, err)
+
+	serv.Block = newTestBlockState(t, testGenesisHeader)
+	serv.Storage = newTestStorageState(t)
+	ch := make(chan *types.Header, 3)
```

Comment:
`Service.Start()` creates a `BlockState` and `StorageState` so I think you can remove lines 151-152 and use `serv.Block` and `serv.Storage` directly